src/components/Tool/GoogleImport/GoogleImport.tsx

```diff
@@ -129,41 +120,97 @@ const GoogleImport: React.FC<Props> = ({ accountListId }: Props) => {
   const { data, loading } = useGoogleContactGroupsQuery();
   const googleAccounts = data?.googleAccounts;
 
-  const account = useMemo(() => {
-    return googleAccounts?.find((account) => account?.id === selectedAccount);
-  }, [googleAccounts, selectedAccount]);
+  const selectedAccount = useMemo(() => {
+    return googleAccounts?.find(
+      (selectedAccount) => selectedAccount?.id === selectedAccountId,
+    );
+  }, [googleAccounts, selectedAccountId]);
 
   const initialValues = useMemo(() => {
     const initialGroupTags =
-      account?.contactGroups.reduce((acc, group) => {
-        if (group?.id) {acc[group?.id] = [group?.tag];}
+      selectedAccount?.contactGroups.reduce((acc, group) => {
+        if (group?.id) {
+          acc[group?.id] = [group?.tag];
+        }
         return acc;
       }, {}) || {};
     return {
       tagsForAllList: [],
       override: 'false',
-      importAll: 'false',
+      importByGroup: 'true',
       groupTags: initialGroupTags,
       groups: [] as string[],
     };
-  }, [account]);
+  }, [selectedAccount]);
 
   useEffect(() => {
-    googleAccounts?.length && setSelectedAccount(googleAccounts[0]?.id || '');
+    googleAccounts?.length && setSelectedAccountId(googleAccounts[0]?.id || '');
   }, [googleAccounts]);
 
   const handleCloseModal = () => {
     setShowModal(false);
-    window.location.href = `${process.env.SITE_URL}/accountLists/${accountListId}/tools`;
     setRedirecting(true);
+    window.location.href = `/accountLists/${accountListId}/tools`;
   };
 
   const onSubmit = async (
     attributes: Attributes,
     { resetForm },
   ): Promise<void> => {
+    setShowConfirmAllModal(false);
+    const session = await getSession();
+    const apiToken = session?.user?.apiToken;
+
+    const groupTags = Object.entries(attributes.groupTags).reduce(
+      (result, value: [string, string[]]) => {
+        result[value[0]] = value[1].join(',');
```

Comment:
Optional, but here's a place you could use array destructuring. I think it's a little clearer. There are probably even better variable names than `key/value`. Is `groupId` and `tags` accurate?

```suggestion
      (result, [key, value]: [string, string[]]) => {
        result[key] = value.join(',');
```